errors/error.go

```diff
@@ -27,12 +27,27 @@ const (
 	// DBOpen indicates a database open error.
 	DBOpen = ErrorKind("ErrDBOpen")
 
+	// DBOpen indicates a database close error.
+	DBClose = ErrorKind("ErrDBClose")
+
 	// DBUpgrade indicates a database upgrade error.
 	DBUpgrade = ErrorKind("ErrDBUpgrade")
 
 	// PersistEntry indicates a database persistence error.
 	PersistEntry = ErrorKind("ErrPersistEntry")
 
+	// CreateTable indicates a database table creation error.
+	CreateTable = ErrorKind("CreateTable")
```

Comment:
CreateTable is specific to SQL and does not apply to Bolt. As we are aiming to make our DB interface agnostic of the implementation, the interface should not include implementation specific errors. 

This also applies to some existing errors - eg. `ErrBucketCreate`  does not apply to SQL, only Bolt.

We should combine these into something more generic, perhaps something like `ErrCreateStorage`